control/control_grpc_server_test.go

```diff
@@ -325,20 +291,24 @@ func TestGRPCServerScheduler(t *testing.T) {
 			So(len(reply.Errors), ShouldEqual, 0)
 			// content to pass to publisher
 			content = reply.Content
-
+			c.UnsubscribeDeps("my-snowflake-id")
 		})
 	})
 	//publishmetrics
 	Convey("PublishMetrics", t, func() {
-		Convey("Should error with invalid inputs", func() {
-			req := controlproxy.GetPubProcReq("snap.gob", []byte{}, "bad name", 1, map[string]ctypes.ConfigValue{}, "my-snowflake-id")
-			reply, err := client.PublishMetrics(context.Background(), req)
-			// we don't expect rpc errors
-			So(err, ShouldBeNil)
-			So(len(reply.Errors), ShouldNotEqual, 0)
-
-			So(reply.Errors[0], ShouldResemble, "bad key")
-		})
+		req := &rpc.SubscribeDepsRequest{
+			Requested: []*common.Metric{&common.Metric{Namespace: common.ToNamespace(fixtures.ValidMetric.Namespace()), Version: int64(fixtures.ValidMetric.Version())}},
+			Plugins: common.ToCorePluginsMsg([]core.Plugin{
+				corePlugin{typeName: "collector", name: "mock", version: 1},
+				corePlugin{typeName: "processor", name: "passthru", version: 1},
+				corePlugin{typeName: "publisher", name: "mock-file", version: 3},
+			},
+			),
+			TaskId:  "my-snowflake-id",
+			CfgTree: common.ToCfgTree(cdata.NewTree()),
+		}
+		_, err := client.SubscribeDeps(context.Background(), req)
+		So(err, ShouldBeNil)
```

Comment:
@IRCody `SubscribeDeps()` is needed to subscribe plugins and create needed pool, otherwise `availablePlugins.processMetrics()` will return with `bad key` error. Same thing applies for publishing. 
Removed blocks dealt with invoking process/publish with invalid inputs, which in current implementation makes no sense, because in such case `SubscribeDeps()` will return with error earlier.